Remove docs URLs from code fence comments

ptgott opened the pull request:
Some example configurations include docs URLs in comments. Since the docs linter does not check whether these correspond to actual docs pages, they can easily go out of date when we reorganize the documentation.

Further, since these docs URLs are absolute, the docs engine does not rewrite them to include version-specific URL paths, so they always link to the default version of the docs.

Some of the changes modify example Terraform resources since we generate the Terraform provider reference.

ptgott committed:
Remove docs URLs from code fence comments

Some example configurations include docs URLs in comments. Since the
docs linter does not check whether these correspond to actual docs
pages, they can easily go out of date when we reorganize the
documentation.

Further, since these docs URLs are absolute, the docs engine does not
rewrite them to include version-specific URL paths, so they always link
to the default version of the docs.

Some of the changes modify example Terraform resources since we generate
the Terraform provider reference.

ravicious commented on docs/pages/includes/config-reference/auth-service.yaml:
Do we need some linter to make sure that those links won't be reintroduced? E.g. something forbidding the use of `#.*goteleport.com/docs` within the `docs` folder.

I was also thinking that we have a lot of links to the docs within the project itself.

```
$ rg "goteleport.com/docs/\w" -g '!docs' -g '!.github' | wc -l
     371
```

A huge chunk of them isn't shown in the UI, those links are meant for Teleport devs. But there's probably a non-insignificant portion of them that's actually shown in the UI. I know that in Connect we do show them in a couple of places.

I almost wonder if we should have some project-wide linter that just checks all strings that start with `goteleport.com/docs/`.

ptgott replied:
We already have a linter preventing `goteleport.com/docs` links in docs pages outside of code snippets, but nothing for code snippets. For code, there's a [script](https://github.com/gravitational/teleport/blob/master/docs/check-redirects/README.md) for checking for necessary docs redirects based on docs URLs in an arbitrary directory path, but no project-wide linter. I can look into improving our linting for docs URLs!

ptgott committed:
Respond to ravicious feedback

- Add context to comment re: the default installer script
- Add context to the configuration reference in places where we removed
  in-comment docs URLs

ptgott commented on docs/pages/includes/config-reference/auth-service.yaml:
```diff
@@ -2,9 +2,6 @@ teleport:
     # audit log. Several back-end types are supported.
-    # See the "Storage backends" (https://goteleport.com/docs/setup/reference/backends)
-    # section of the documentation to learn how to configure
-    # DynamoDB, S3, etcd, and other highly available back-ends.
```
I've restored links in "Further reading" lists in the configuration reference.
